dev/ColorPicker/ColorSpectrum.cpp

```diff
@@ -177,12 +181,39 @@ void ColorSpectrum::OnKeyDown(winrt::KeyRoutedEventArgs const& args)
     // The order of saturation and value in the spectrum is reversed - the max value is at the bottom while the min value is at the top -
     // so we want left and up to be lower for hue, but higher for saturation and value.
     // This will ensure that the icon always moves in the direction of the key press.
-    const IncrementDirection direction =
+    IncrementDirection direction =
         (incrementChannel == winrt::ColorPickerHsvChannel::Hue && (args.Key() == winrt::VirtualKey::Left || args.Key() == winrt::VirtualKey::Up)) ||
         (incrementChannel != winrt::ColorPickerHsvChannel::Hue && (args.Key() == winrt::VirtualKey::Right || args.Key() == winrt::VirtualKey::Down)) ?
         IncrementDirection::Lower :
         IncrementDirection::Higher;
 
+    // Image is flipped in RightToLeft, so we need to invert direction in that case.
+    if (FlowDirection() == winrt::FlowDirection::RightToLeft &&
+        (args.Key() == winrt::VirtualKey::Left || args.Key() == winrt::VirtualKey::Right))
+    {
+        if (direction == IncrementDirection::Higher)
+        {
+            direction = IncrementDirection::Lower;
+        }
+        else
+        {
+            direction = IncrementDirection::Higher;
+        }
+    }
+
+    // The SaturationValue configuration is also inverted, so invert to ensure correct left right navigation
```

Comment:
>SaturationValue [](start = 11, length = 15)

If its both saturation value and RTL we should switch back right? since we are double inverted?

Reply:
Yes, with RTL, we would invert it twice resulting in the initial value. We could try to guard against double inverting, but I  think we would end up with an if case that's hard to understand.

Reply:
Nevermind, the if case is not that complex after all, updated it now.